### PR TITLE
Add better error messages for unavailable swarm commands

### LINKED ISSUE #23845: This node is not participating as a Swarm manager
Hey there

Referring to 1.12 and SwarmKit here:

I'm getting 
`This node is not participating as a Swarm manager`
upon issuing docker node ls or docker swarm inspect.

Is there a reason that a non-manager node cannot inspect the swarm or list the nodes in the cluster? Is it just not implemented yet?

Thanks!
Alessandro

### PR DESCRIPTION
**Pull request by tonistiigi**
Add more descriptive error messages to swarm control commands depending on the current state of the node.

fixes #23845

Signed-off-by: Tonis Tiigi tonistiigi@gmail.com

### CODE REVIEW

**Comment by vdemeester** [comment]
LGTM 🐯

File: daemon/cluster/cluster.go [hunk 1]
```diff
@@ -629,13 +626,25 @@ func (c *Cluster) isActiveManager() bool {
...
 }
 
+// errNoManager returns error describing why manager commands can't be used.
+// Call with read lock.
+func (c *Cluster) errNoManager() error {
+	if c.node == nil {
+		return fmt.Errorf("This node is not participating as a Swarm manager. Use \"docker swarm init\" or \"docker swarm join --manager\" to connect this node to Swarm and try again.")
+	}
+	if c.node.Manager() != nil {
+		return fmt.Errorf("This node is not participating as a Swarm manager. No active connection to the control API.")
```
Comment (aaronlehmann) [inline daemon/cluster/cluster.go]: s/participating as //

The "No active connection to the control API." part is a little confusing. It's more like a debugging message than an actionable error. I'd suggest removing it or replacing it with something that explains the condition.

File: daemon/cluster/cluster.go [hunk 2]
```diff
@@ -629,13 +626,25 @@ func (c *Cluster) isActiveManager() bool {
...
+	}
+	return fmt.Errorf("This node is not participating as a Swarm manager. Worker nodes can't be used to view or modify cluster state. Please run this command on a manager node or promote the current node to a manager.")
```
Comment (aaronlehmann) [inline daemon/cluster/cluster.go]: s/participating as //

File: daemon/cluster/cluster.go (hunk 1 above)
Comment (tonistiigi) [inline daemon/cluster/cluster.go]: "Manager is being prepared or has trouble connecting to the cluster." ? Yes, this isn't really actionable(unless you want to leave the cluster) but thats why it's important to separate it from the cases where user just forgot to type a command.

Comment (aaronlehmann) [inline daemon/cluster/cluster.go]: That sounds much better, thanks. 👍

**Comment by aaronlehmann** [comment]
LGTM

**Pull request merged**